File: TriviaServer/TriviaServer/LoginRequest.cpp
#include "LoginRequest.h"

//-------------constructor--------------------
LoginRequest::LoginRequest(nlohmann::json j)
{
	this->_username = j.at("username").get<string>();
	this->_password = j.at("password").get<string>();
}

/*
This function returns the password
Input:None
Output:The username ::string
*/
string LoginRequest::getUsername()
{
	return this->_username;
}
/*
This function returns the password
Input:None
Output:The password ::string
*/
string LoginRequest::getPassword()
{
	return  this->_password;
}

File: TriviaWPFClient/TriviaClient/LoginRequest.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Newtonsoft.Json;
using Newtonsoft.Json.Bson;

namespace TriviaClient
{
    public class LoginRequest : Request
    {
        public string username { get; set; }
        public string password { get; set; }

        public LoginRequest(string _username, string _password):base(20)
        {
            this.username = _username;
            this.password = _password;
        }

        public string GetUsername()
        {
            return this.username;
        }

        public string GetPassword()
        {
            return this.password;
        }

        public override byte[] CastToBson()
        {
            MemoryStream ms = new MemoryStream();
            using (BsonWriter writer = new BsonWriter(ms))
            {
                JsonSerializer serializer = new JsonSerializer();
                serializer.Serialize(writer, this);
            }
            return ms.ToArray();
        }
    }
}
